### PR TITLE
Assign random seed for lmi dist

### PR DESCRIPTION
**Pull request by sindhuvahinis**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java [hunk 1]
```diff
@@ -123,7 +128,7 @@ public Output addInput(Input input, int timeout) throws TranslateException {
                     throw new TranslateException("Time out in: " + timeout);
                 }
             }
-            Request req = new Request(input);
+            Request req = new Request(input, Long.toUnsignedString(random.nextLong()));
```
Comment (lanking520) [inline engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java:131]: https://github.com/vllm-project/vllm/blob/main/vllm/utils.py#L39 use UUID here

File: engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java [hunk 2]
```diff
@@ -147,12 +152,14 @@ private static final class Request {
...
         boolean last;
+        String seed;
 
-        Request(Input input) {
+        Request(Input input, String seed) {
             this.input = input;
             data = new ChunkedBytesSupplier();
             output = new Output();
             output.add(data);
+            this.seed = seed;
```
Comment (lanking520) [inline engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java:162]: why seed is a string?

Comment (sindhuvahinis) [reply engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java:162]: For serialization and deserialization.

Comment (lanking520) [reply engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java:162]: then this is not a seed? Why we need seed for serialize and deserialize

Comment (lanking520) [reply engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java:164]: uuid4

File: engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java [hunk 3]
```diff
@@ -48,11 +50,13 @@ class RollingBatch implements Runnable {
     private ReentrantLock lock;
     private Condition canAdd;
     private Condition canRead;
+    private Random random;
 
     RollingBatch(PyProcess process, int maxRollingBatchSize, int timeout) {
         this.process = process;
         this.maxRollingBatchSize = maxRollingBatchSize;
         this.timeout = timeout;
+        this.random = new Random();
```
Comment (frankfliu) [inline engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java:59]: Use RandomUtils

**Review by frankfliu — APPROVED** [review]

File: engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java (hunk 3 above)
Comment (sindhuvahinis) [reply engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java]: done

File: engines/python/setup/djl_python/huggingface.py [hunk 4]
```diff
@@ -224,6 +224,12 @@ def inference(self, inputs):
                         "In order to enable dynamic batching, all input batches must have the same parameters"
                     )
 
+            seed_key = 'seed' if inputs.get_batch_size() > 1 else f'batch_{i}.seed'
+            if item.contains_key(seed_key):
```
Comment (frankfliu) [inline engines/python/setup/djl_python/huggingface.py:228]: should we pop 'seed' from parameters before inference?

File: engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java [hunk 5]
```diff
@@ -123,7 +129,7 @@ public Output addInput(Input input, int timeout) throws TranslateException {
                     throw new TranslateException("Time out in: " + timeout);
                 }
             }
-            Request req = new Request(input);
+            Request req = new Request(input, Long.toUnsignedString((long) RandomUtils.random()));
```
Comment (frankfliu) [inline engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java:132]:
```suggestion
            Request req = new Request(input, String.valueOf(RandomUtils.nextInt()));
```

Comment (sindhuvahinis) [reply engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java]: done

File: engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java [hunk 6]
```diff
@@ -123,7 +129,7 @@ public Output addInput(Input input, int timeout) throws TranslateException {
                     throw new TranslateException("Time out in: " + timeout);
                 }
             }
-            Request req = new Request(input);
+            Request req = new Request(input, Integer.toString(RandomUtils.nextInt()));
```
Comment (frankfliu) [inline engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java]:
```suggestion
            Request req = new Request(input, String.valueOf(RandomUtils.nextInt()));
```

**Review by frankfliu — APPROVED** [review]

**Pull request merged**